src/txn/txn_recover.c

```diff
@@ -503,6 +504,23 @@ __recovery_set_ckpt_base_write_gen(WT_RECOVERY *r)
     return (ret);
 }
 
+/*
+ * __recovery_txn_setup_initial_state --
+ *     Setup the transaction initial state required by rollback to stable.
+ */
+static int
+__recovery_txn_setup_initial_state(WT_SESSION_IMPL *session, WT_RECOVERY *r)
```

Comment:
I think this is not entirely correct. This will only run with the case for recovery not live database. We need to update the timestamp and txn id for that as well.

Reply:
Other than recovery, the `pinned timestamp` is set in different flows whenever the `oldest timestamp` is set by the application. Only in recovery we set the `oldest timestamp` directly without an API, so we need to set the `pinned timestamp` also.

Reply:
I see. You are keeping the update in both places. Thanks. This is all good